### PR TITLE
refactor(core): Fix type errors in workflow, core, nodes-langchain, and nodes-base (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
In the spirit of today's cleanup sessions, this PR cleans up type errors on a couple of backend packages, and also deletes a lot of old dead code in tests.

## Review / Merge checklist
- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/nodes-base/nodes/Discord/test/v2/node/channel/get.test.ts [hunk 1]
```diff
@@ -5,10 +5,11 @@ import { getResultNodeData, setup, workflowToTests } from '@test/nodes/Helpers';
 import type { WorkflowTestData } from '@test/nodes/types';
 import { executeWorkflow } from '@test/nodes/ExecuteWorkflow';
 
+// TODO: use nock
```
Comment (ivov) [inline packages/nodes-base/nodes/Discord/test/v2/node/channel/get.test.ts:8]: Maybe we should get into the habit of creating a story for each TODO and adding the ID on the TODO line.

File: packages/nodes-base/nodes/Google/Drive/test/v2/node/drive/get.test.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ describe('test GoogleDriveV2: drive get', () => {
 		jest.unmock('../../../../v2/transport');
 	});
 
-	it('shuold be called with', async () => {
+	it('should be called with', async () => {
```
Comment (ivov) [inline packages/nodes-base/nodes/Google/Drive/test/v2/node/drive/get.test.ts:29]: We should consider a `grammarcheck` script...

File: packages/workflow/test/AugmentObject.test.ts [hunk 3]
```diff
@@ -290,7 +290,7 @@ describe('AugmentObject', () => {
 		});
 
 		test('should work with complex values on first level', () => {
-			const originalObject = {
+			const originalObject: any = {
```
Comment (ivov) [inline packages/workflow/test/AugmentObject.test.ts:293]: I take it there is no other way?

**Review by ivov — APPROVED** [review]

File: packages/nodes-base/nodes/Discord/test/v2/node/channel/get.test.ts (hunk 1 above)
Comment (netroy) [reply packages/nodes-base/nodes/Discord/test/v2/node/channel/get.test.ts:8]: https://linear.app/n8n/issue/ENG-117

File: packages/workflow/test/AugmentObject.test.ts (hunk 3 above)
Comment (netroy) [reply packages/workflow/test/AugmentObject.test.ts:293]: in this case, unfortunately not. it's not worth defining a complex type here just for the test.

**Pull request merged**